orc8r/lib/go/service/service.go

```diff
@@ -133,33 +139,84 @@ func NewServiceWithOptionsImpl(moduleName string, serviceName string, serverOpti
 // Run the service. This function blocks until its interrupted
 // by a signal or until the gRPC server is stopped.
 func (service *Service) Run() error {
-	port, err := registry.GetServicePort(service.Type)
-	if err != nil {
-		return fmt.Errorf("get service port: %v", err)
-	}
+	wg := new(sync.WaitGroup)
+	wg.Add(2)
+	errChan := make(chan error)
+	go func() {
+		errChan <- service.run()
+		wg.Done()
+	}()
+	go func() {
+		errChan <- service.runProtected()
+		wg.Done()
+	}()
+	wg.Wait()
```

Comment:
I don't think this logic is quite right -- consider if 1 of the servicers returns an err and the other continues to run. Then only one of the wg.Done will fire, so the wg.Wait will never complete, so the 1 err will never be returned from this func

Lmk if I'm misreading

Reply:
I think the wg has to be there for the two services to run simultaneously, otherwise two go routines will just fire and then it will proceed with the main go routine and exit. 

Or I could do something like 
```
go func () {
    service.runProtected()
}

service.run() 
```
And that would be equivalent I believe, because the last line would block the main go routine
Was loosely referencing this https://medium.com/rungo/running-multiple-http-servers-in-go-d15300f4e59f

Though it does only return one error, I changed it so that it returns both.